src/Orchard.Cms.Web/Modules/Orchard.Setup/Views/Setup/Index.cshtml

```diff
@@ -1,8 +1,32 @@
 ﻿@using Orchard.Setup.Controllers
+@using Microsoft.AspNetCore.Builder
+@using Orchard.Utility
 @inject Microsoft.AspNetCore.Mvc.Localization.IViewLocalizer T
+@inject Microsoft.Extensions.Options.IOptions<IdentityOptions> identityOptions
 @model Orchard.Setup.ViewModels.SetupViewModel
 @{
     var defaultRecipe = Model.Recipes.FirstOrDefault(x => x.Name == Model.RecipeName) ?? Model.Recipes.First();
+    List<string> passwordOptions = new List<string>();
+    IdentityOptions options = identityOptions.Value;
+    if (options.Password.RequireNonAlphanumeric)
+    {
+        passwordOptions.Add(T["one non-alphanumeric"].Value);
+    }
+    if (options.Password.RequireUppercase)
+    {
+        passwordOptions.Add(T["one uppercase"].Value);
+    }
+    if (options.Password.RequireLowercase)
+    {
+        passwordOptions.Add(T["one lowercase"].Value);
+    }
+    if (options.Password.RequireDigit)
+    {
+        passwordOptions.Add(T["one digit"].Value);
+    }
+    passwordOptions.Add(string.Format(T["{0} characters"].Value, options.Password.RequiredLength));
+
+    string passwordTooltip = string.Format(T["Password must have at least {0}."].Value, string.Join(T[", "].Value, passwordOptions.ToArray()).ReplaceLastOccurrence(T[", "].Value, T[" and "].Value));
```

Comment:
This is wrong, whenever you'll try to call `T.Value` you can be sure there will be double encoding issues.
Here, just create an `HtmlBuilder` and append the `IHtmlString` values returned by the `T` calls. Don't try to call Join either as it doesn't handle `IHtmlContent`.